dotCMS/src/main/java/com/dotcms/security/secret/SecretsStoreKeyStoreImpl.java

```diff
@@ -0,0 +1,348 @@
+     * @return
+     */
+    private KeyStore saveSecretsStore(final KeyStore keyStore) {
+        final File secretStoreFile = new File(secretsKeyStorePath);
+        final File secretStoreFileTmp = new File(secretStoreFile.getParent(), "dotSecretsStore_" + System.currentTimeMillis() + ".p12.tmp");
+
+        secretStoreFileTmp.getParentFile().mkdirs();
+        try (OutputStream fos = Files.newOutputStream(secretStoreFileTmp.toPath())) {
+            keyStore.store(fos, loadStorePassword());
+            FileUtil.copyFile(secretStoreFileTmp, secretStoreFile);
+            secretStoreFileTmp.delete();
+        } catch (Exception e) {
+            Logger.warn(this.getClass(), "unable to save secrets store " + secretStoreFileTmp + ": " + e);
+            throw new DotRuntimeException(e);
+        }
+        return keyStore;
+    }
+
+    @VisibleForTesting
+    protected final static char[] CACHE_404 = new char[] {'4', '0', '4'};
+
+    /**
+     * Verifies if the key exists in the store
+     * @param variableKey
+     * @return
+     */
+    public boolean containsKey(final String variableKey){
+        try {
+            return getKeysFromCache().contains(variableKey);
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    /**
+     * secret values are stored encryped in cache for 30 seconds and are decrypted when requested. If
+     * value is not in cache, it will be read from the keystore
+     * 
+     * @param variableKey
+     * @return
+     * @throws Exception
+     */
+    public Optional<char[]> getValue(final String variableKey) {
+
+        final char[] fromCache = getFromCache(variableKey,
+                () -> loadValueFromStore(variableKey));
+
+        return Arrays.equals(fromCache, CACHE_404) ? Optional.empty() : Optional.ofNullable(decrypt(fromCache));
+
+    }
+
+    /**
+     * tries to load the value from the store on disk
+     * 
+     * @param variableKey
+     * @return
+     */
+    private char[] loadValueFromStore(final String variableKey) {
+        try {
+           final KeyStore keyStore = getSecretsStore();
+            if (keyStore.containsAlias(variableKey)) {
+                final PasswordProtection keyStorePP = new PasswordProtection(loadStorePassword());
+                final SecretKeyFactory factory = SecretKeyFactory.getInstance(SECRETS_STORE_SECRET_KEY_FACTORY_TYPE);
+                final SecretKeyEntry secretKeyEntry = (SecretKeyEntry) keyStore.getEntry(variableKey, keyStorePP);
+                return ((PBEKeySpec) factory.getKeySpec(secretKeyEntry.getSecretKey(), PBEKeySpec.class)).getPassword();
+            } else {
+                return CACHE_404;
+            }
+
+        } catch (Exception e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+
+
+    @Override
+    public Set<String> listKeys() {
+        return Sneaky.sneak(this::getKeysFromCache);
+    }
+
+    @Override
+    public synchronized boolean deleteAll(){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use block level rather than method level synchronization](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)